Project/Source/Components/UI/ComponentText.h

```diff
@@ -21,19 +21,30 @@ class ComponentText : public Component {
 	void Load(JsonValue jComponent) override;				// Deserializes
 	void DuplicateComponent(GameObject& owner) override;	// Duplicates Component
 	
-	void Draw(ComponentTransform2D* transform) const;				// Draws the text ortographically using the active camera and the position of the Tranform2D. It will apply the color as tint
+	void Draw(const ComponentTransform2D* transform) const;				// Draws the text ortographically using the active camera and the position of the Tranform2D. It will apply the color as tint
 	TESSERACT_ENGINE_API void SetText(const std::string& newText);	// Sets text
 	TESSERACT_ENGINE_API void SetFontSize(float newfontSize);		// Sets fontSize
 	TESSERACT_ENGINE_API void SetFontColor(const float4& newColor); // Sets color
 	float4 GetFontColor() const;									// Returns Color
-	void RecalculcateVertices();									// Recalculate verticesText. This is called when Text/Font/Transform is modified in order to recalculate the position of vertices.
+	void RecalculcateVertices();									// Recalculate verticesText. This is called when Text/Font/FontSize/LineHeight/Transform is modified in order to recalculate the position of vertices. Will calculate the position based on the horizontal Text Alignment
 
 private:
+	float SubstringWidth(const char* substring, float scale);		// Returns the advanced width of the substring until it reaches the end of line or new line character ('\0' or '\n').
+
+private:
+	enum class TextAlignment {
+		LEFT,
+		CENTER,
+		RIGHT
+	};
+
 	std::string text = "Text";									   // Text to display
 	std::vector<std::array<std::array<float, 4>, 6>> verticesText; // Vertices per each character
 
-	float fontSize = 12.0f;		// Font size
-	float4 color = float4::one; // Color of the font
+	float fontSize = 12.0f;									// Font size
```

Comment:
El valor por defecto podría ser más grande? Con valor 12 por defecto los textos se ven super chiquititos, es una tontería porque a priori todo el mundo querrá cambiar el tamaño del texto a su gusto pero igual tener un valor por defecto algo más visible a primera vista da un mejor feel

Reply:
Changed to 24. Though it's no longer fontSize in Pixels since the new scale is applied.